cgsmiles/resolve.py

```diff
@@ -69,62 +68,80 @@ def match_bonding_descriptors(source, target, bond_attribute="bonding"):
 
 class MoleculeResolver:
     """
-    Resolve the molecule described by a CGBigSmile
-    string.
+    Resolve the molecule(s) described by a
+    CGSmiles string. Each execution of the
+    resolve function will return the next
+    resolved molecule and the graph of the
+    previous resolution.
+
+    Use the resolve_iter method to loop over
+    all possible molecule resolutions enconded
+    in the CGSmiles string.
     """
-
     def __init__(self,
                  pattern,
                  meta_graph=None,
                  fragment_dict={},
-                 all_atom=True):
+                 last_all_atom=True):
 
-        # let's start by setting some attributes
-        # this is the fragment string
-        self.fragment_str = None
-        # this is the cgsmiles string
-        self.cgsmiles_str = None
+        """
+        Parameters
+        ----------
+        pattern: str
+            the cgsmiles string to resolve
+        meta_graph: `:class:nx.Graph`
+            a potential higher resolution graph
+        fragment_dict: dict[`:class:nx.Garph`]
+            a dict of fragment graphs
+        last_all_atom: bool
+            if the last resolution is at the all
+            atom level. Default: True
```

Comment:
This is relevant because it determines when to switch to pysmiles right? So have the docstring say that.
Does this also mean I cannot mix resolutions? I.e. a layer that has both some regular atoms as well as some CG ones? 
Is there no decent way to have pysmiles /try/ to parse it, and if it fails assume it's not atomistic?

Reply:
Yes, that's what it means. You wouldn't mix layers anyway. I cannot think of a single situation where that would make sense and not introduce logic errors. Once we established that, only the last layer can be atomistic. I prefer to keep it more explicit via the flag. It's a little price to pay in terms of API and I'm not a fan of a general except everything statement. 

I will adjust the docstring.